Libraries/LibGUI/GTextEditor.cpp

```diff
@@ -563,6 +563,15 @@ void GTextEditor::keydown_event(GKeyEvent& event)
                 did_update_selection();
             }
         }
+        else if (m_cursor.line() == line_count() - 1) {
+            int new_column = current_line().length();
+            toggle_selection_if_needed_for_event(event);
+            set_cursor(m_cursor.line(), new_column);
+            if (m_selection.start().is_valid()) {
```

Comment:
This block is the same in all three branches. We can move it below and only have one copy if it, before the `return`.